k-distribution/include/kframework/builtin/domains.md

```diff
@@ -835,13 +835,13 @@ module SET-KORE-SYMBOLIC [symbolic,haskell]
   rule S              -Set .Set           => S          [simplification]
   rule .Set           -Set  _             => .Set       [simplification]
   rule SetItem(X)     -Set (S SetItem(X)) => .Set
-                               ensures notBool (X in S) [simplification]
+                               ensures notBool (X in S) [simplification, preserves-definedness]
   rule S              -Set (S SetItem(X)) => .Set
-                               ensures notBool (X in S) [simplification]
+                               ensures notBool (X in S) [simplification, preserves-definedness]
   rule (S SetItem(X)) -Set S              => SetItem(X)
-                               ensures notBool (X in S) [simplification]
+                               ensures notBool (X in S) [simplification, preserves-definedness]
```

Comment:
Similar argument as for the `Map` cases above, the set `S` could be undefined for other reasons than containing `X`...